### DISCUSSION TITLE
Why do I need to "Merge into current branch"?

### DISCUSSION THREAD

**Discussion by danielb987** [community/community#21866] [New to GitHub]
<p>I’m working on a Java library:<br>
<a href="https://github.com/danielb987/JavaDiptraceAsciiLib" rel="nofollow noopener">https://github.com/danielb987/JavaDiptraceAsciiLib</a></p>
<p>Currently I’m the only contributor and all contributions are done to my branch “Daniel”. Every time I do a Pull request from the branch “Daniel” to the master and then merges the changes into the master, I need to do a “Merge into current branch” from master to Daniel, in order to have my branch “Daniel” up to date.</p>
<p><a href="https://github.com/danielb987/JavaDiptraceAsciiLib/branches" rel="nofollow noopener">https://github.com/danielb987/JavaDiptraceAsciiLib/branches</a></p>
<p>Why do I need to merge my changes back to my branch “Daniel”? Every changes to the master branch is already made to the branch “Daniel”.</p>
<p>I’m using GitHub Desktop for git operations.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>When you merge the pull request into <code>master</code> there’s one commit that you’re missing in <code>Daniel</code>: the one titled “Merge pull request #[number] from [branch name]”. Basically, the standard GitHub Pull Request flow is designed for branches that only live as long as the PR does. Since you are using a long-lived branch model, you have to merge that one commit back each time.</p>
